tests/units/providers/test_openshift_provider.py

```diff
@@ -0,0 +1,241 @@
+        op.oapi_resources = ['foo']
+        op.openshift_artifacts = {
+            'pods': [
+                {
+                    'metadata': {
+                        'namespace': 'foo'
+                    }
+                }
+            ]
+        }
+
+        op.deploy()
+
+        self.mock_oc.deploy.assert_called_once_with(
+            'namespaces/foo/pods/?access_token=None',
+            op.openshift_artifacts['pods'][0])
+
+    def test_deploy_dryrun(self):
+        op = self.get_oc_provider(dryrun=True)
+        op.oapi_resources = ['foo']
+        op.openshift_artifacts = {
+            'pods': [
+                {
+                    'metadata': {
+                        'namespace': 'foo'
+                    }
+                }
+            ]
+        }
+
+        op.deploy()
+
+        self.assertFalse(self.mock_oc.deploy.call_count)
+
+
+class TestOpenshiftProviderProcessArtifactData(OpenshiftProviderTestMixin, unittest.TestCase):
+
+    def test_process_artifact_data_non_template_kind(self):
+        artifact_data = {
+            'kind': 'Pod',
+            'pods': [
+                {
+                    'metadata': {
+                        'namespace': 'foo'
+                    }
+                }
+            ]
+        }
+        self.mock_oc.get_oapi_resources.return_value = ['pods']
+
+        op = self.get_oc_provider()
+
+        op._process_artifact_data('foo', artifact_data)
+
+        self.assertEqual(op.openshift_artifacts,
+                         {'pod': [artifact_data]})
+
+    def test_process_artifact_data_template_kind(self):
+        artifact_data = {
+            'kind': 'Template',
+            'objects': [
+                {
+                    'kind': 'Pod',
+                    'metadata': {
+                        'namespace': 'foo'
+                    }
+                },
+                {
+                    'kind': 'Service',
+                    'metadata': {
+                        'namespace': 'foo'
+                    }
+                }
+            ]
+        }
+        self.mock_oc.get_oapi_resources.return_value = ['templates']
+        op = self.get_oc_provider()
+        self.mock_oc.process_template.return_value = artifact_data['objects']
+
+        op._process_artifact_data('foo', artifact_data)
+
+        self.assertEqual(
+            op.openshift_artifacts, {
+                'pod': [
+                    {'kind': 'Pod', 'metadata': {'namespace': 'foo'}}
+                ],
+                'service': [
+                    {'kind': 'Service', 'metadata': {'namespace': 'foo'}}
+                ]
+            }
+        )
+
+    def test_process_artifact_data_error_resource_not_in_resources(self):
+        artifact_data = {
+            'kind': 'foobar'
+        }
+
+        op = self.get_oc_provider()
+
+        self.assertRaises(
+            ProviderFailedException,
+            op._process_artifact_data, 'foo', artifact_data)
```

Comment:
No, this is how ``self.assertRaises`` is supposed to be called.

``self.assertRaises(ExceptionClass, method, arg1, arg2)``